services/QuillLMS/app/models/translated_text.rb

```diff
@@ -4,19 +4,21 @@
 #
 # Table name: translated_texts
 #
-#  id                 :bigint           not null, primary key
-#  locale             :string           not null
-#  translation        :text
-#  created_at         :datetime         not null
-#  updated_at         :datetime         not null
-#  english_text_id    :integer          not null
-#  translation_job_id :string           not null
+#  id              :bigint           not null, primary key
+#  locale          :string           not null
+#  source_api      :string
+#  translation     :text             not null
+#  created_at      :datetime         not null
+#  updated_at      :datetime         not null
+#  english_text_id :integer          not null
 #
 class TranslatedText < ApplicationRecord
+  GENGO_SOURCE = "gengo"
+  OPEN_AI_SOURCE = "open_ai"
   belongs_to :english_text
-  scope :pending_translation, -> { where(translation: nil) }
-
-  def self.fetch_and_save_pending! = pending_translation.each(&:fetch_translation!)
-
-  def fetch_translation! = Gengo::SaveTranslatedText.run(translation_job_id)
+  scope :ordered_by_source_api, lambda { |source_api = OPEN_AI_SOURCE|
```

Comment:
While our codebase has some [cases](https://github.com/empirical-org/Empirical-Core/pull/9288) of Arel, those involved a requirement to get around order clauses using raw sql, we tend to avoid its use.

Also, I think the current query is susceptible to SQL injection since the string is being interpolated directly.  Using '?'  placeholder instead with plain SQL would be preferred here: 
```
scope :ordered_by_source_api, lambda { |source_api = OPEN_AI_SOURCE|
  order("CASE WHEN source_api = ? THEN 0 ELSE 1 END, source_api ASC", source_api)
}
```

Reply:
Hah, this is what I originally had, and I think Rubocop changed it! I will switch it back

Reply:
Hmm, I can't do your version because rspec complains: 
```
ActiveRecord::UnknownAttributeReference:
       Dangerous query method (method whose arguments are used as raw SQL) called with non-attribute argument(s): "CASE WHEN source_api = ? THEN 0 ELSE 1 END, source_api ASC".This method should not be called with user-provided values, such as request parameters or model attributes. Known-safe values can be passed by wrapping them in Arel.sql().
```

We can sanitize ourselves like this: 
```
scope :ordered_by_source_api, lambda { |source_api = OPEN_AI_SOURCE|
    source_api = OPEN_AI_SOURCE unless SOURCES.include? source_api
    order (
      Arel.sql("CASE WHEN source_api = '#{source_api}' THEN 0 ELSE 1 END, source_api ASC")
    )
  }
```

Reply:
Let's add a 
```
SOURCE_APIS = [GENGO_SOURCE, OPEN_AI_SOURCE]

validates :source_api, inclusion: { in: SOURCE_APIS }
```